bots/psuteam7bot/communication.js

```diff
@@ -58,6 +58,73 @@ communication.initTeamCastleInformation = (self) => {
     }
 }
 
+/**
+ * Checks whether target tile is visible and is empty OR occupied and not a castle and report to allied castles if so
+ * Returns the x-coord or y-coord value of the target tile depending on the map reflection
+ *
+ */
+communication.checkAndReportEnemyCastleDestruction = (self) => {
+    const {x, y} = self.target;
+    const robotID = self.getVisibleRobotMap()[y][x];
+
+    //Case, empty target, castle is destroyed
+    if(robotID === 0)
+    {
+        //Add coords to pending messages, push y then x
+        self.pendingMessages.push(y);
+        self.pendingMessages.push(x);
+        return true;
+    }
+    else //Case target occupied or not in visible radius, -1 or there is a robotID > 0
+    {
+        //Check if robot is not a castle, if so, report castle destruction
+        if(robotID > 0 && self.getRobot(robotID).unit !== 0)
+        {
+            //Add coords to pending messages, push y then x
+            self.pendingMessages.push(y);
+            self.pendingMessages.push(x);
+            return true;
+        }
+    }
+    return false;
+}
 
+communication.sendCastleTalkMessage = (self) => {
+    if(self.pendingMessages.length > 0)
```

Comment:
I like the idea of a queue, but we need to be careful about stale information (might be more relevant on the castle/reading side, but I'm thinking about it here)

Reply:
Castle reading only uses the value for removing from enemyCastle list and then only checks whether the removed item is the castle's current target. Otherwise, it just costs time to check through enemyCastle list for stale/ duplicate message.